doc/system-telemetry/reboot-cause.md

```diff
@@ -0,0 +1,116 @@
+# Reboot-cause information via telemetry agent
+
+## Revision
+
+| Rev | Date     | Author      | Change Description |
+|:---:|:--------:|:-----------:|--------------------|
+| 0.1 | 09/02/20 | Sujin Kang  | Initial version    |
+
+## Scope
+-Enable sonic streaming telemetry agent to send Reboot-cause information
+
+### Enable sonic streaming telemetry agent to send Reboot-cause information
+
+#### Part 1
+During the boot, the process-reboot-cause processes the last reboot-cause based on the hardware reboot-cause
+and the software reboot-cause information and creates previous-reboot-cause.txt with the information as it does currently.
+In addition to save the history of the previous reboot-cause, `process-reboot-cause` will save the previous
+reboot cause information to "/host/reboot-cause/previous-reboot-cause/" with adding timestamp at the end of file name.
+And the file will be formatted to be parsed easily.
+
+The example shows the previous reboot-cause files stored in /host/reboot-cause/previous-reboot-cause/.
+```
+$ls /host/reboot-cause/previous-reboot-cause/
+previous-reboot-cause-20200903T232033.txt
+previous-reboot-cause-20200902T101105.txt
+previous-reboot-cause-20200902T015048.txt
+...
+```
+The following example shows the content of the previous reboot-cause file - previous-reboot-cause-20200903T232033.txt.
+```
+[
+    {
+        gen_time: "20200903T232033",
+        cause: "reboot",
+        user: "admin",
+        time: "Thu 03 Sep 2020 11:15:30 PM UTC",
+        comment: ""
+    }
+]
+```
+
+#### Part 2
+A new service named as `post-process-reboot-cause.service` which will retrieve the saved reboot-cause files and read each reboot-cause information from the files
```

Comment:
If we go with this approach, we might want to consider naming the first service "determine-reboot-cause" and the second service "process-reboot-cause", or similar. It's more meaningful than "post-process-reboot-cause".